src/compiler/scala/tools/reflect/ReflectGlobal.scala

```diff
@@ -11,6 +12,12 @@ import scala.tools.nsc.Settings
 class ReflectGlobal(currentSettings: Settings, reporter: Reporter, override val rootClassLoader: ClassLoader)
   extends Global(currentSettings, reporter) with scala.tools.reflect.ReflectSetup with scala.reflect.runtime.SymbolTable {
 
+  override lazy val analyzer = new {
+    val global: ReflectGlobal.this.type = ReflectGlobal.this
+  } with Analyzer {
+    override lazy val macroParentClassLoader = rootClassLoader
+  }
+
```

Comment:
Thanks for the quick feedback!

My rational was that the default behaviour of `findMacroClassloader` to use the class loader of Macros as parent felt rather arbitrary to me (it might break in weird ways if somebody decides to load scala-compiler in an osgi-ish way). The public documentation of [`defaultMacroClassloader`](https://github.com/scala/scala/blob/2.11.x/src/compiler/scala/reflect/macros/runtime/MacroRuntimes.scala#L39-L47) dispatches to `findMacroClassloader`, which is undocumented in this respect. I can of course improve on this. Following the proposed approach perhaps `ReplGlobal` should also just override `macroParentClassLoader` and leave `findMacroClassloader` as is.

On the other hand I can understand, why exposing further API takes caution. I can't tell which aspect to prefer, so please let me know a decision on this and I'll update the pull request.